app/controllers/concerns/admin_only_maintenance_mode.rb

```diff
@@ -0,0 +1,21 @@
+module AdminOnlyMaintenanceMode
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :redirect_non_admins_in_maintenance_mode
+  end
+
+private
+
+  def redirect_non_admins_in_maintenance_mode
+    return unless Settings.features.maintenance_mode.enabled
+
+    flash[:notice] = Settings.features.maintenance_mode.message
```

Comment:
Not sure about this. It's going to end up popping up for admins on each page when maintenance mode is on. Actually that's probably not a big deal, I'll have a look what it's like on the review app.